rest_framework/routers.py

```diff
@@ -35,6 +35,12 @@
 DynamicListRoute = namedtuple('DynamicListRoute', ['url', 'name', 'initkwargs'])
 
 
+def replace_curly_brackets(url_path):
+    if ('{' and '}') in url_path:
+        url_path = url_path.replace('{', '{{').replace('}', '}}')
+    return url_path
```

Comment:
Okay, I think this is valid enough, yup, tho let's make a couple of minor tweaks:

1. `escape_curly_brackets` would be slightly more clear.
2. Let's include a docstring noting that we escape any brackets in the regex for the purposes of string formatting.